### DISCUSSION TITLE
How can I "just draw" with `Printer`?

### DISCUSSION THREAD

**Discussion by stnbu** [gyscos/cursive#723] [Q&A]
I may be using the wrong approach or the wrong framework, but I'm just tinkering and really having fun with Cursive. I'm a fan. I also don't know what I'm doing yet.

I got the crazy idea of implementing Tetris. I have low exceptions: I'd be happy with just colored squares.

I was hoping to just use `Printer`, but I don't know how to make that a parameter of a callback, since I've only seen it so far in the signature of a `with_draw(F)`

```rust
use cursive::theme::{Color, ColorStyle};
use cursive::view::Resizable;
use cursive::views::Canvas;
use cursive::Cursive;
use cursive::Printer;

const WIDTH: usize = 10;
const HEIGHT: usize = 20;

#[derive(Copy, Clone, Debug)]
enum Cell {
    Empty,
    Red,
    Blue,
}

impl From<Cell> for Color {
    fn from(cell: Cell) -> Color {
        match cell {
            Cell::Empty => Color::Rgb(0, 0, 0),
            Cell::Red => Color::Rgb(255, 0, 0),
            Cell::Blue => Color::Rgb(0, 0, 255),
        }
    }
}

struct Board([[Cell; WIDTH]; HEIGHT]);

impl Default for Board {
    fn default() -> Self {
        Board([[Cell::Empty; WIDTH]; HEIGHT])
    }
}

fn main() {
    let mut siv = cursive::default();
    siv.set_user_data(Board::default());
    siv.add_layer(Canvas::new(()).with_draw(draw).fixed_size((HEIGHT, WIDTH)));
    siv.add_global_callback('x', |s| test_draw_func(s));
    siv.run();
}

fn test_draw_func(c: &mut Cursive) {
    // I would like a copy of `&Printer` too, so I can do some arbitrary x,y drawing.
    let _ = c
        .with_user_data(|board: &mut Board| {
            for row in board.0 {
                for cell in row {
		    // Was hoping to color a square using "Cell"
                    drop(cell);
                }
            }
        })
        .unwrap();
}
```

How can I nudge this along?

**Comment by stnbu** [comment]
Some more context, maybe to help guide me if you're willing.

I was going to use user input (keyboard) to manipulate the contents of the `Board` and then hopefully print the whole thing out on a timer.

**Comment by esotericbyte** [comment]
I am also learning about this library. Have you looked at colors.rs example? 
It shows how to print an x y  grid with a printer and  with each block a different color.
I have been finding features in the docs then use grep the examples for them. 
I was really happy last week when I discovered "bat" a replacement for cat/less in rust that shows beautiful syntax highlighted example code in the terminal! Very cool.
